index.d.ts

```diff
@@ -291,102 +291,52 @@ export interface KeyInfoProvider {
    * @param keyInfo - An optional array of XML Nodes.
    * @return A string or Buffer representing the key.
    */
-  getKey(keyInfo?: Node[]): string | Buffer;
+  getCert(keyInfo?: Node[]): string | Buffer;
 
   /**
    * Method to return an XML string representing the contents of a KeyInfo element.
    *
-   * @param key - An optional string representing the key.
+   * @param cert - An optional string, or array of strings, representing the certificate to be included in the KeyInfo element.
    * @param prefix - An optional string representing the namespace alias.
    * @return An XML string representation of the contents of a KeyInfo element.
    */
-  getKeyInfo(key?: string, prefix?: string): string;
+  makeKeyInfo(cert?: string | string[], prefix?: string): string;
 
   /**
    * An optional dictionary of attributes which will be added to the KeyInfo element.
    */
-  attrs?: { [key: string]: string };
+  attrs?: { [cert: string]: string };
 }
 
-/**
- * The FileKeyInfo class loads the certificate from the file provided in the constructor.
- */
-export class FileKeyInfo implements KeyInfoProvider {
+export class KeyInfo implements KeyInfoProvider {
   /**
-   * The path to the file from which the certificate is to be read.
+   * The certificate  to be used for signing.
```

Comment:
Misleading. The private key is used for signing, the certificate is used for validating the signature.

Reply:
You're statement is true. I will review the code comments.